File: PC^2/pc2admin/samps/sumit.java
import java.io.*;

//
// File:    sumit.java
// Purpose: to sum the integers in the file sumit.dat
// Author:  pc2@ecs.csus.edu or http://www.ecs.csus.edu/pc2
// 
// $Id: sumit.java,v 1.2 2005/03/06 05:19:05 laned Exp $
//

public class sumit {
    public static void main(String[] args) 
    {
	try
	{	
		RandomAccessFile file = new RandomAccessFile("sumit.dat", "r");
		String line;
		int sum = 0;
		int rv = 0;
		while((line = file.readLine()) != null) 
		{			
			rv = new Integer(line.trim()).intValue();
			if (rv > 0)
				sum = sum + rv;
			// System.out.println(line);
		}
		System.out.print("The sum of the integers is ");
		System.out.println(sum);
	}
	catch(Exception e)
	{
		System.out.println("Possible trouble reading sumit.dat");
	}			
    }
}

// eof sumit.java $Id: sumit.java,v 1.2 2005/03/06 05:19:05 laned Exp $

File: PC^2/pc2admin/samps/sumit.pas


{
*
* File:    sum1.pas
* Purpose: prints the sum of the positive integers in sumit.dat
* Author:  pc2@ecs.csus.edu or http://www.ecs.csus.edu/pc2
*
* $Id: sumit.pas,v 1.1 2005/03/06 05:13:57 laned Exp $
*
}

var
  tf  : text;
  i   : integer;
  sum : integer;

begin
   assign (tf,'sumit.dat');
   {$i-} reset (tf); {$i+}
   if ioresult <> 0 then
      writeln('could not open file sumit.dat ')
   else
   begin

   sum := 0;
   while not eof (tf) do
   begin
       readln(tf,i);
       if (i > 0) then
            sum := sum + i;
   end;

   writeln('The sum of the positive integers is ',sum);
   end;

end.
{ EOF }
